rsmtool/utils/commandline.py

```diff
@@ -230,117 +253,620 @@ def existing_configuration_file(string):
     return parser
 
 
-def generate_configuration(context,
-                           use_subgroups=False,
-                           as_string=False,
-                           suppress_warnings=False):
+class InteractiveField:
     """
-    Automatically generate an example configuration for a given
-    command-line tool.
+    Class that encapsulates a configuration field that is computed interactively.
 
-    Parameters
+    Attributes
+    ----------
+    choices : list
+        List of possible choices for the field value if ``data_type`` is "choice".
```

Comment:
Just for consistency
```suggestion
        List of possible choices for the field value if ``data_type`` is ``choice``.
```